scripts/check_one_single_file_test.sh

```diff
@@ -18,47 +18,63 @@ set -e
 set -u
 set -x
 
-if [ -z "${DREDD_SKIP_CHECK_COMPILE_COMMANDS+x}" ]
+DREDD_INSTALLED_EXECUTABLE=${DREDD_REPO_ROOT}/third_party/clang+llvm-13.0.1/bin/dredd
+
+# Move to the temporary directory
+cd "${DREDD_REPO_ROOT}"
+cd temp/
+
+if [ -z "${DREDD_SKIP_COPY_EXECUTABLE+x}" ]
 then
-  DREDD_INSTALLED_EXECUTABLE=${DREDD_REPO_ROOT}/third_party/clang+llvm-13.0.1/bin/dredd
-  # Move to the temporary directory
-  cd "${DREDD_REPO_ROOT}"
-  cd temp/
   # Ensure that Dredd is in its installed location. This depends on a
   # debug build being available
   cp build-Debug/src/dredd/dredd ${DREDD_INSTALLED_EXECUTABLE}
-  f="${DREDD_REPO_ROOT}/${1}"
-  # Copy the single-file test case to the temporary directory so
-  # that it can be mutated without affecting the original
-  cp $f .
-  copy_of_f=$(basename $f)      
-  # Mutate the test case using Dredd with optimisations
-  ${DREDD_INSTALLED_EXECUTABLE} --mutation-info-file temp.json ${copy_of_f} --
-  # Check that the mutated test case is as expected
-  diff ${copy_of_f} $f.expected
-  # Check that the mutated file compiles
-  if [[ $f == *.cc ]]
+fi
+
+f="${DREDD_REPO_ROOT}/${1}"
+
+for opt in "opt" "noopt"
+do
+  DREDD_EXTRA_ARGS=""
+  DREDD_EXPECTED_FILE=""
+  if [ ${opt} == "noopt" ]
   then
-    ${CXX} -c ${copy_of_f}
+    DREDD_EXTRA_ARGS="--no-mutation-opts"
+    DREDD_EXPECTED_FILE="$f.noopt.expected"
   else
-    ${CC} -c ${copy_of_f}
+    DREDD_EXPECTED_FILE="$f.expected"
   fi
+
   # Copy the single-file test case to the temporary directory so
   # that it can be mutated without affecting the original
   cp $f .
   copy_of_f=$(basename $f)
-  # Mutate the test case using Dredd without optimisations
-  ${DREDD_INSTALLED_EXECUTABLE} --no-mutation-opts --mutation-info-file temp.json ${copy_of_f} --
-  # Check that the mutated test case is as expected
-  diff ${copy_of_f} $f.noopt.expected
+
+  # Mutate the test case using Dredd with optimisations
+  ${DREDD_INSTALLED_EXECUTABLE} ${DREDD_EXTRA_ARGS} --mutation-info-file temp.json ${copy_of_f} --
```

Comment:
This comment shouldn't be here anymore because the command will also run Dredd without optimisations.

Reply:
Fixed, thanks.